#166: Added support for the interface name in the config file

tank0nf opened the pull request:
Hey @glaslos,

Reffering to issue #166 
I've made the necessary changes including adding new field to the configuration file for specifying the interface name.
Here’s how you can use the command:
```
sudo bin/server -i <interface name>
```
Example use case: `sudo bin/server -i wlan0`

By default, it will use `eth0`.

Additionally, I've included instructions for installing and running the project on Fedora. Since the `bin/server` command requires sudo privileges, I’ve also tried upgrading the README.md file to reflect these changes accordingly.

I would greatly appreciate your feedback on any changes or further modifications needed, I will be glad to do so!
Thank you,
Tank0nf.

glaslos commented on Makefile:
```diff
@@ -30,7 +32,7 @@ clean:
 
 run: build
-	sudo bin/server -i eth0
+	sudo bin/server -i $(INTERFACE)
```
You don't need to pass the interface here if we get it from the config

glaslos commented on README.md:
````diff
@@ -16,14 +16,19 @@ Arch:
 
+Fedora:
+```
+sudo dnf install gcc libpcap-devel iptables
````
nice, thank you!

glaslos commented on config/config.yaml:
```diff
@@ -1,3 +1,8 @@
+interface: eth0
+tcpPort: 8080
+udpPort: 8081
+sshPort: 2222
```
why are you adding these ports?

glaslos commented on glutton.go:
```diff
@@ -40,9 +41,18 @@ type Glutton struct {
 //go:embed config/rules.yaml
 var defaultRules []byte
 
+func init() {
```
don't use init(). initConfig is the correct place.

glaslos commented on glutton.go:
```diff
@@ -58,6 +68,12 @@ func (g *Glutton) initConfig() error {
 	viper.SetDefault("rules_path", "rules/rules.yaml")
+	viper.SetDefault("interface", "eth0") // Default interface name
+
+	if interfaceName == "" {
```
this should not be necessary

glaslos commented on glutton.go:
```diff
@@ -108,7 +124,7 @@ func New(ctx context.Context) (*Glutton, error) {
 func (g *Glutton) Init() error {
 	var err error
-	g.publicAddrs, err = getNonLoopbackIPs(viper.GetString("interface"))
+	g.publicAddrs, err = getNonLoopbackIPs(interfaceName)
```
use viper.GetString("interface")

glaslos commented on glutton.go:
```diff
@@ -241,11 +257,11 @@ func (g *Glutton) Start() error {
 
 	sshPort := viper.GetUint32("ports.ssh")
-	if err := setTProxyIPTables(viper.GetString("interface"), g.publicAddrs[0].String(), "tcp", uint32(g.Server.tcpPort), sshPort); err != nil {
+	if err := setTProxyIPTables(interfaceName, g.publicAddrs[0].String(), "tcp", uint32(g.Server.tcpPort), sshPort); err != nil {
```
use viper.GetString("interface")

tank0nf commented:
Hello @glaslos,

Thank you for your feedback! I’ve made the necessary changes and am looking forward to contributing further to this project.

Please feel free to let me know if any additional adjustments are needed, I’d be happy to make them.

Thanks again,
Tank0nf.

glaslos commented on Makefile:
```diff
@@ -3,6 +3,7 @@ NAME := glutton
 BUILDSTRING := $(shell git log --pretty=format:'%h' -n 1)
 VERSIONSTRING := $(NAME) version $(VERSION)+$(BUILDSTRING)
 BUILDDATE := $(shell date -u -Iseconds)
+CONFIG_FILE=config/config.yaml
```
why do you need this?

glaslos commented on glutton.go:
```diff
@@ -52,12 +57,16 @@ func (g *Glutton) initConfig() error {
 	}
 	// If no config is found, use the defaults
-	viper.SetDefault("ports.tcp", 5000)
+	viper.SetDefault("ports.ttcp", 5000)
```
why? please review your own PR

glaslos commented on glutton.go:
```diff
@@ -52,12 +57,16 @@ func (g *Glutton) initConfig() error {
 	viper.SetDefault("rules_path", "rules/rules.yaml")
+	viper.SetDefault("interface", "eth0") // Default interface name
+
+	interfaceName = viper.GetString("interface")
```
And what do you do with this variable?

tank0nf commented:
Hello @glaslos,

Thank you for your feedback! I’ve made the necessary changes and am looking forward to contributing further to this project.

Please feel free to let me know if any additional adjustments are needed, I’d be happy to make them.

Thanks again,
Tank0nf.

glaslos commented on glutton.go:
```diff
@@ -44,6 +45,10 @@ type Glutton struct {
 var defaultRules []byte
 
 func (g *Glutton) initConfig() error {
+	var interfaceName string
+	flag.StringVar(&interfaceName, "i", "", "Network interface name")
+	flag.Parse()
```
why are you doing this?

tank0nf commented:
Hi @glaslos,
I have made the necessary changes, let me know if their is any more issue.
Thank you,
Tank0nf.

glaslos commented:
Now the last change missing is in the Makefile

tank0nf commented:
Hi @glaslos ,
I’ve made the necessary updates to the Makefile. Could you please review it and let me know if any further adjustments are needed?

Thank you,
Tank0nf

glaslos commented on Makefile:
```diff
@@ -30,7 +30,7 @@ clean:
 
 run: build
-	sudo bin/server -i eth0
+	sudo bin/server -c config.yaml
```
It is not necessary to define the config path. Also I don't think this will work since the config file is in a folder.

tank0nf replied:
Hi @glaslos,
I have added an if statement to check for the interface details, please let me know if their is any other changes necessary.

tank0nf commented:
Hi @glaslos 
I have added an if statement to check for the interface details, please let me know if their is any other changes necessary.
Please let me know if their is any further changes.
Thank you,
Tank0nf.

glaslos commented on Makefile:
```diff
@@ -30,7 +30,12 @@ clean:
+	else \
+		INTERFACE=eth0; \
+	fi; \
+	sudo bin/server -c config.yaml -i $$INTERFACE
```
why do you need this? We have defaults for all the flags. The interface should not be defined in the Makefile. If someone wants to change the interface during the development, they would do it in the config file.

tank0nf commented:
Hi @glaslos,
I have removed the config file thing and just reverted to 
```
run: build
	sudo bin/server
```
If their is any further changes please let me know. I would be glad to follow through!
Thank you,
Tank0nf.

glaslos reviewed (approved):
LGTM. Thank you
